SFR/outer_fuel_pincell/make_SFR_pincell_model.py

```diff
@@ -0,0 +1,113 @@
+    settings = openmc.Settings()
+    settings.source = openmc.IndependentSource(space=uniform_dist)
+    settings.batches = 200
+    settings.inactive = 40
+    settings.particles = 20000
+
+    if shannon_entropy:
+        entropy_mesh = openmc.RegularMesh()
+        entropy_mesh.lower_left = lower_left
+        entropy_mesh.upper_right = upper_right
+        entropy_mesh.dimension = (10, 10, 20)
+        settings.entropy_mesh = entropy_mesh
+
+    settings.temperature = {
+        "default": 280.0 + 273.15,
+        "method": "interpolation",
+        "range": (294.0, 3000.0),
+        "tolerance": 1000.0,
+    }
+    settings.export_to_xml()
+
+
+def main():
+    ap = ArgumentParser(description="SFR Pincell Model Generator")
+    ap.add_argument(
+        "-n",
+        dest="n_axial",
+        type=int,
+        default=1,
+        help="Number of cells in the Z direction",
+    )
+    ap.add_argument(
+        "-e",
+        "--shannon_entropy",
+        action="store_true",
+        help="Add Shannon entropy mesh"
+    )
+
+
+    arguments = ap.parse_args()
+    arguments = ap.parse_args()
+
+    N = arguments.n_axial
+    shannon_entropy = arguments.shannon_entropy
+    height = geom.height
+
+    outer_fuel_material = make_sfr_material(material_dict['outer_fuel'], percent_type='wo')
+    cladding_material = make_sfr_material(material_dict['cladding'], percent_type='ao')
+    sodium = make_sfr_material(material_dict['sodium'], percent_type='ao')
+    helium = make_sfr_material(material_dict['helium'], percent_type='ao')
+    materials = openmc.Materials(
+        [outer_fuel_material, sodium, helium, cladding_material]
+    )
+    materials.export_to_xml()
+
+    fuel_or = openmc.ZCylinder(r=geom.r_fuel)
+    clad_ir = openmc.ZCylinder(r=geom.r_clad_inner)
+    clad_or = openmc.ZCylinder(r=geom.r_clad_inner+geom.t_clad)
+
+    z_plane = [openmc.ZPlane(z0=i) for i in np.linspace(-height / 2, height / 2, N + 1)]
+
+    top = z_plane[-1]
+    bottom = z_plane[0]
+    top.boundary_type = "vacuum"
+    bottom.boundary_type = "vacuum"
+
+    inner_cells = {"fuel": [], "gas_gap": [], "clad": [], "sodium": []}
+    all_inner_cells = []
+
+    for i in range(N):
+
+        layer = +z_plane[i] & -z_plane[i + 1]
+        inner_cells["fuel"].append(
+            openmc.Cell(fill=outer_fuel_material, region=layer & -fuel_or)
+        )
+        inner_cells["gas_gap"].append(
+            openmc.Cell(fill=helium, region=layer & +fuel_or & -clad_ir)
+        )
+        inner_cells["clad"].append(
+            openmc.Cell(fill=cladding_material, region=layer & +clad_ir & -clad_or)
+        )
+        inner_cells["sodium"].append(openmc.Cell(fill=sodium, region=+clad_or & layer))
+
+        all_inner_cells.extend(
+            [
+                inner_cells["fuel"][i],
+                inner_cells["gas_gap"][i],
+                inner_cells["clad"][i],
+                inner_cells["sodium"][i],
+            ]
+        )
```

Comment:
See [this](https://github.com/neams-th-coe/cardinal_amr/blob/221a7f18b78729e39b9b7bf48e035ed427009a56/models/lwr/pincell/make_openmc_model.py#L37) in the LWR models for a cleaner solution